src/main/java/seedu/typists/Main.java

```diff
@@ -1,39 +1,41 @@
 package seedu.typists;
 
-import seedu.typists.exception.InvalidStringInputException;
-import seedu.typists.parser.StringParser;
+import seedu.typists.parser.Parser;
+import seedu.typists.ui.TextUi;
 
-import java.util.List;
-import java.util.Scanner;
 
+//solution below adapted from https://github.com/se-edu/addressbook-level2/
 public class Main {
+    /** Version info of the program. */
+    public static final String VERSION = "Typist - Version 1.0";
+    TextUi uiBot;
+    Parser parseBot;
+
+    public void start() {
+        this.uiBot = new TextUi();
+        this.parseBot = new Parser();
+        uiBot.showWelcomeMessage(VERSION);
+    }
+
+    public void runCommandLoopUntilExitCommand() {
+
+    }
+
+    public void exit() {
+
+    }
+
+    /** Runs the program until termination.  */
+    public void run() {
+        start();
+        runCommandLoopUntilExitCommand();
+        exit();
+    }
+
     /**
      * Main entry-point for the java.duke.Duke application.
      */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-        System.out.println("What is your name?");
-
-        Scanner in = new Scanner(System.in);
-        System.out.println("Hello " + in.nextLine());
-
-        /**
-         * The following code test on the JUnit test on StringParser
-         */
-        StringParser sp = new StringParser();
-        String str = "Hello|World|How are|you all";
-        List<String> stringParts = null;
-        try {
-            stringParts = sp.splitString(str, "\\|");
-        } catch (InvalidStringInputException e) {
-            e.printStackTrace();
-        }
+        new Main().run();
```

Comment:
Why start running this way? Do consider putting the lines of code in run() to the main method instead